Improve a few steps configuring SSO on Azure (#1767)
neo4j/docs-operations: modules/ROOT/pages/tutorial/tutorial-sso-configuration.adoc

This example shows how to configure Entra ID for authentication and authorization using an access token.

> . Assuming that you have already created and registered your SSO application in Azure, you should now configure a token in Entra ID using the *Token configuration* tab.

Author: What is SPA?

> . Assuming that you have already created and registered your SSO application in Azure, you should now configure a token in Entra ID using the *Token configuration* tab.

Author: I looked through this page, and I was under impression that all these guidances don't depend upon each other, they are independent. If you need instructions on how to configure Entra ID using an access token, you won't go to the section about Okta or to the next section on ID token. In fact, this is not one tutorial, but a few tutorials. Do you want to tie them up?

> . Assuming that you have already created and registered your SSO application in Azure, you should now configure a token in Entra ID using the *Token configuration* tab.

Reviewer: No, I don't want to tie them up. I want to make them consistent in style, structure, and words.

> . Assuming that you have already created and registered your SSO application in Azure, you should now configure a token in Entra ID using the *Token configuration* tab.

Reviewer: Ah, I have removed it. It means Single-page application.

. Next step is to go to the **Expose an API** tab and click the **Add a Scope** button. 
+
You need to know your Essentials located in the GUID -- Globally Unique Identifier. 
+

> .The GUID is used to identify specific resources or instances within Azure. You can find it on the app registration page.

Author: I think it's better to keep the caption in terms of consistency. All other screenshots on the page have their captions

image::sso-configuration-tutorials/azure-id.svg[]
+

> If you click the **Add a Scope** button for the first time, you see a new window stating that you need to add an _Application ID URI_ before proceeding:

Author: why do you want to remove this step but suggest to add the line 'Save and continue'?

> +

Reviewer: ```suggestion
```

[source]
----

> api://4376dc8b-b5af-424f-9ada-c1c1b2d416b9/access-token

Reviewer: I think the screenshot is enough. We don't need to have it in a code format as well as it is different for every app.

----

> . Next fill in all mandatory fields in the panel **Add a scope** and copy your scope which will be used for _neo4j.conf_ file.

Reviewer: ```suggestion
. Fill in all mandatory fields in the window **Add a scope** and make a note of them.
```

> . Configure Neo4j by adjusting the following settings in the _neo4j.conf_ file: 

Reviewer: ```suggestion
. Configure Neo4j to use Entra ID for authentication by configuring the following settings in the _neo4j.conf_ file: 
```

> .. Set parameters to be `access_token`:

Reviewer: ```suggestion
```

+
[source, properties]
----

> dbms.security.oidc.azure.config=principal=unique_name;code_challenge_method=S256;token_type_principal=access_token;token_type_authentication=access_token

Reviewer: ```suggestion
# Configure the access_token
dbms.security.oidc.azure.config=principal=unique_name;code_challenge_method=S256;token_type_principal=access_token;token_type_authentication=access_token
# Configure the OIDC token endpoint with the Directory (tenant) ID
dbms.security.oidc.azure.token_endpoint=https://login.microsoftonline.com/54e85725-ed2a-49a4-a19e-11c8d29f9a0f/oauth2/v2.0/token
# Configure the iss claim in the id token with the Directory (tenant) ID
# Make sure you add the trailing slash (`/`) at the end of the URL, or this operation might fail.
dbms.security.oidc.azure.issuer=https://sts.windows.net/54e85725-ed2a-49a4-a19e-11c8d29f9a0f/
# Provide the Entra ID parameters, such as client_id, response_type, scope, etc.
dbms.security.oidc.azure.params=client_id=4376dc8b-b5af-424f-9ada-c1c1b2d416b9;response_type=code;scope=openid profile email api://4376dc8b-b5af-424f-9ada-c1c1b2d416b9/access-token
```

Reviewer: I would suggest adding all these configurations in one code block so that they can copy and paste them and then amend the values as per their app instead of having separate steps for each line.

----

> .. Add the following parameter:

Reviewer: ```suggestion
```

> .. Add the following parameter:

Reviewer: ```suggestion
```

+

> [source, properties]

Reviewer: ```suggestion
```

----
dbms.security.oidc.azure.token_endpoint=https://login.microsoftonline.com/54e85725-ed2a-49a4-a19e-11c8d29f9a0f/oauth2/v2.0/token
----

> .. Include the issuer:

Reviewer: ```suggestion
```

> +

Reviewer: All these need to be deleted.

This example shows how to configure Entra ID for authentication and authorization using an access token.

. Assuming that you have already created and registered your SSO application in Azure, you should now configure a token in Entra ID using the *Token configuration* tab.

. Next step is to go to the **Expose an API** tab and click the **Add a Scope** button. 
+
You need to know your Essentials located in the GUID -- Globally Unique Identifier. 
+
.The GUID is used to identify specific resources or instances within Azure. You can find it on the app registration page.
image::sso-configuration-tutorials/azure-id.svg[]

> The GUID is used to identify specific resources or instances within Azure.

Reviewer 2: ```suggestion
```

> You can find it on the app registration page:

Reviewer 2: ```suggestion
```

This example shows how to configure Entra ID for authentication and authorization using an access token.

. It is assumed that you have already created and registered your SSO application in Azure.

. Configure a token by navigating to the **Token configuration** tab.

> . To configure Neo4j, you need to know your Essentials.

Author: Hi! No, this is not a part of the previous step on how to configure a token. Actually, you need essentials for the next step, to expose API and add a scope. So, probably it's a good idea to incorporate this info into the next step.

> . Configure a token by navigating to the **Token configuration** tab.

Author: ```suggestion
```

This example shows how to configure Entra ID for authentication and authorization using an access token.

. After the successful creation of your SSO application in Azure, open the *Token configuration* tab to configure a token.
.. In the Manage section, click *Token configuration* and then *Add groups claim*.
.. Select *Security groups* to include in your access token.
.. Save your changes.

. Next step is to open the **Expose an API** tab and click the **Add a Scope** button.

> ..If you click the **Add a Scope** button for the first time, you see a new window stating that you need to add an _Application ID URI_ before proceeding

Author: ```suggestion
.. If you click the **Add a Scope** button for the first time, you see a new window stating that you need to add an _Application ID URI_ before proceeding
```

You can find it on your app *Overview* page. 
+
.The GUID is used to identify specific resources or instances within Azure. You can find it on the app registration page.
image::sso-configuration-tutorials/azure-id.svg[]
+

> Click *Save and continue*.

Author: ```suggestion
```

> . Fill in all mandatory fields in the window **Add a scope** and make a note of them.

Author: ```suggestion
.. Fill in all mandatory fields in the window **Add a scope** and make a note of them.
.. Click *Save and continue*.
```

This example shows how to configure Entra ID for authentication and authorization using an access token.

. After the successful creation of your SSO application in Azure, open the *Token configuration* tab to configure a token.
.. Click *Add groups claim*.
.. Select *Security groups* to include in your access token.
.. Save your changes.

. Open the **Expose an API** tab and select **Add a Scope**.
.. If you click the **Add a Scope** button for the first time, you see a new pane stating that you need to add an _Application ID URI_ before proceeding.
+
You can find it on your app *Overview* page. 
+
.The GUID is used to identify specific resources or instances within Azure. You can find it on the app registration page.
image::sso-configuration-tutorials/azure-id.svg[]
+

> Click *Save and continue* after setting the _Application ID URI_.

Author: ```suggestion
.. Click *Save and continue* after setting the _Application ID URI_.
```

+
[source, properties]
----
# Configure the access_token
dbms.security.oidc.azure.config=principal=unique_name;code_challenge_method=S256;token_type_principal=access_token;token_type_authentication=access_token
# Configure the OIDC token endpoint with the Directory (tenant) ID
dbms.security.oidc.azure.token_endpoint=https://login.microsoftonline.com/54e85725-ed2a-49a4-a19e-11c8d29f9a0f/oauth2/v2.0/token
# Configure the iss claim in the id token with the Directory (tenant) ID
# Make sure you add the trailing slash (`/`) at the end of the URL, or this operation might fail.
dbms.security.oidc.azure.issuer=https://sts.windows.net/54e85725-ed2a-49a4-a19e-11c8d29f9a0f/
# Provide the Entra ID parameters, such as client_id, response_type, scope, etc.
dbms.security.oidc.azure.params=client_id=4376dc8b-b5af-424f-9ada-c1c1b2d416b9;response_type=code;scope=openid profile email api://4376dc8b-b5af-424f-9ada-c1c1b2d416b9/access-token
----
+
[NOTE]
====
As previously mentioned, the GUID here is also the Directory (tenant) ID.
Make sure you add the trailing slash (`/`) at the end or this operation might fail.

> The audience parameter for access tokens are typically set with` api://` at the front.

Reviewer: ```suggestion
The audience parameter for access tokens is typically set with `api://` at the front.
```

> 

Reviewer 2: ````suggestion
````

image::sso-configuration-tutorials/azure-id.svg[]

> ====

Reviewer 2: ````suggestion
````

+
[NOTE]
====
As previously mentioned, the GUID here is also the Directory (tenant) ID.
Make sure you add the trailing slash (`/`) at the end or this operation might fail.
====

.. Include the following statement:
+
[source, properties]
----
dbms.security.oidc.azure.params=client_id=4376dc8b-b5af-424f-9ada-c1c1b2d416b9;response_type=code;scope=openid profile email api://4376dc8b-b5af-424f-9ada-c1c1b2d416b9/access-token
----

> +

Reviewer: ```suggestion
+
[NOTE]
====
```

> Note that the audience parameter for access tokens are typically set with` api://` at the front.

Reviewer: ```suggestion
The audience parameter for access tokens is typically set with `api://` at the front.

For more information on each of these settings, see xref:configuration/configuration-settings/#_security_settings[Security settings].
====
```
